app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -811,9 +812,13 @@ public void startLoading(final boolean forceLoad) {
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe((@NonNull StreamInfo result) -> {
                     isLoading.set(false);
-                    currentInfo = result;
-                    handleResult(result);
-                    showContent();
+                    if (result.getAgeLimit() != NO_AGE_LIMIT) {
```

Comment:
I don't understand where you are checking if the user has disabled restricted videos

Reply:
Actually, enabling or disabling the option does nothing

Reply:
I forgot to commit that ;)